src/lib.rs

```diff
@@ -837,15 +894,28 @@ impl<'a> RecordBuilder<'a> {
     /// [`Metadata::builder().build()`]: struct.MetadataBuilder.html#method.build
     #[inline]
     pub fn new() -> RecordBuilder<'a> {
-        RecordBuilder {
+        #[cfg(feature = "kv_unstable")]
+        return RecordBuilder {
             record: Record {
                 args: format_args!(""),
                 metadata: Metadata::builder().build(),
                 module_path: None,
                 file: None,
                 line: None,
+                key_values: KeyValues(&Option::None::<(key_values::Key, key_values::Value)>),
```

Comment:
For the extremely common case of no key-values, I wonder whether there would be a measurable performance difference from using a (private) Source impl whose visit fn does nothing, unlike Option's which needs to check for Some every time.

Reply:
This sounds like a good excuse to bootstrap some micro-benchmarks for the `log` crate, so I'll come back to this one later!